crates/types/src/lib.rs

```diff
@@ -856,29 +865,105 @@ impl TypeTrace for Global {
     }
 }
 
-/// Globals are initialized via the `const` operators or by referring to another import.
-#[derive(Debug, Clone, Copy, Hash, Eq, PartialEq, Serialize, Deserialize)]
-pub enum GlobalInit {
-    /// An `i32.const`.
+/// A constant expression.
+///
+/// These are used to initialize globals, table elements, etc...
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct ConstExpr {
+    ops: SmallVec<[ConstOp; 8]>,
```

Comment:
8 seems pretty big here given how often we'll be storing this, could this perhaps be tuned to be the native size of `SmallVec`?

Reply:
WasmHeapType::Extern

Reply:
Even without a const fn could this perhaps be smaller to start? A `ConstOp` is 32 bytes large and 8 of them inline here is 256 bytes for a single expression which seems pretty egregious especially if a table is initialized with a list of constant expressions.

I'm basically pretty fearful of the impact of using this everywhere when the representation is 8x larger than it is today

Reply:
Yeah I dropped it to 2 locally, just haven't pushed the updates yet until I finish addressing the other comments.